src/spyglass/spikesorting/spikesorting_sorting.py

```diff
@@ -157,11 +157,11 @@ def make(self, key: dict):
                     np.arange(np.searchsorted(timestamps, interval[0]),
                               np.searchsorted(timestamps, interval[1])))
             list_triggers = [list(np.concatenate(list_triggers))]
-            
+            pad = 2 * (1 / fs) * 1000
             recording = sit.remove_artifacts(
                 recording=recording,
                 list_triggers=list_triggers,
-                ms_before=0, ms_after=0, mode='zeros')
```

Comment:
Just curious about the logic here. This is removing 2 ms after the triggering artifact, which I assume is based on some sort of threshold crossing. Do we also want to remove any time before? How was 2 ms chosen?

Reply:
I believe the `pad` should only be 2 time indices and not 2ms since it's being multiplied by `(1/fs)`, but my logic might be wrong on that.

The plots are showing the traces post-artifact zeroing with an amplitude threshold of 1000uV. The red points show where artifacts were zeroed and are also plotted at the uV threshold, so it's visible that no data crosses that threshold. Does that make sense? I can rephrase (or add better labeled plots) if that would be helpful?

Reply:
Ah okay. I think it would be helpful to get comments from people who have dealt with artifacts before. I will try to tag some people.

Reply:
To clarify, we detect every point that crosses the threshold and apply our own user-defined padding. These points then get passed to a spike interface method to zero them out. This pad is just the minimal time required for the SI code to actually zero it. Prior to this, nothing was getting zeroed because the `ms_before` and `ms_after` arguments were 0, and the SI implementation was trying to slice the array from `[n:n]`, which is nothing.

Reply:
Shouldn't this only affect the artifacts that are short in duration? Or is that the only kind of artifact we actually end up seeing in practice?

Reply:
I believe this is relevant line of code from spikeinterface:

https://github.com/SpikeInterface/spikeinterface/blob/d79e6e720b2d4ce0431cce62f049cdb703e21059/spikeinterface/toolkit/preprocessing/remove_artifacts.py#L115-L123

Reply:
I don't think the duration matters given our implementation, since `list_triggers` ends up being a list of a list of sequential indices that cover the entire detected artifact. So regardless of how long the artifact, SI only sees the individual index, not the span. Given the line you linked, if `ms_before` AND `ms_after` are both 0, then it will set nothing to 0 because the slice will be `[n:n]`. For example:

```
>>> arr = np.arange(15)
>>> trig = 10
>>> pad = [0,0]
>>> arr[trig-pad[0]:trig+pad[-1]] = 0
>>> arr
array([ 0,  1,  2,  3,  4,  5,  6,  7,  8,  9, 10, 11, 12, 13, 14])
```

It seems like I should come up with a more concise and clear explanation of what problem this is addressing. I'll add it to this PR.

Reply:
Okay, my confusion is that `_get_artifact_times` includes a `removal_window_ms` argument. So if I understand you correctly, we are just removing each index in that interval individually, which leads to this problem of needing to specify the padding here. If that is the case, that seems rather inefficient and perhaps explains @MichaelCoulter 's issue #243. Am I getting that right?

Reply:
Thanks all for working on this!

@edeno in response to your question, I agree with @dpeg22 that the artifact detection algorithm already builds in some user-defined "padding" or removal window around any data points that cross the artifact detection threshold(s). So here when we get artifact_times it is already an interval list with a bunch of intervals that each already include times covering the artifacts and a small window around them.  Those intervals in the artifact_times interval list vary in duration, depending how long the threshold violations lasted. So now, this ms_before and ms_after is separate from the algorithm that finds those artifact_times intervals, and instead relates to actually removing those intervals from the data, using spikeinterface. However, spikeinterface is not currently handling intervals, and instead expects the list_triggers.

I think it is the case that we only really want to remove anything in the artifact_times intervals, and nothing before or after each interval in artifact_times. However, it seems here that we have translated the artifact_times intervals back to a list of triggers, which spikeinterface creates slices around, to zero out data. If the pre/post trigger removal times are 0 for spikeinterface, there is no "slice" to remove around each artifact point, so nothing gets zero'd out. This is overall sort of an odd way of doing things - to go from a time interval list of detected artifact intervals, to a list of indices, that get turned into inferred times based on sampling rates, that get sliced... or something like that. **I think it could be great if spikeinterface could instead handle an interval or slice of indices** - @khl02007 and I chatted and he is thinking about what we can do here and what we may be able to request from spikeinterface (better yet, would be great if si could handle an interval of times! but that might be too much to ask :) ).

I can add that I am a little confused about how many problems are being discussed at once here though: (1) artifact detection algorithm inefficiency from @MichaelCoulter filed issue (2) axes=0 or =1 in detecting proportion of electrodes with a detected threshold violation that I think @dpeg22 is addressing in this PR and (3) given some detected artifact intervals, when you go to spikesort, how do those artifacts actually get removed with spikeinterface? Which @dpeg22 is also trying to address here I think. Point (3) here is also the only topic my comments are relevant to. Maybe it would be helpful to separate out these different points of discussion.